File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n=nums.length;
        int idx=0;
        for(int i=0;i<n;i++){
            if(nums[i]!=0){
                nums[i]=nums[i]+nums[idx]-(nums[idx]=nums[i]);
                idx++;
            }
        }   
        return ;
    }
}

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        res=[]
        c=0
        for i in nums:
            if i==0:
                c=c+1
            else:
                res.append(i)
        while c>0:
            res.append(0)
            c=c-1
        for i in range(len(nums)):
            nums[i]=res[i]